src/components/LinkEncoder.js

```diff
@@ -22,6 +28,8 @@ import { Message, AccessTime, Numbers, Download } from '@mui/icons-material';
  *   Need to change to a better alignment
  */
 
+let interval;
```

Comment:
This is essential because when calling the pinging function, it will create an timer with setInterval and assign it to the interval variable. If the interval variable is somewhere inside the LinkEncoder object, it will get overwritten every time and there's no way to stop the interval that was made initially. I learned this after spending a whole day and realizing that the interval variable will be overwritten every new render cycle and there's no way to access the previous cycle's interval variable to stop it, causing it to run forever.